src/test/java/teammates/test/cases/testdriver/HtmlHelperTest.java

```diff
@@ -135,4 +135,20 @@ public void testConvertToStandardHtmlPart() throws Exception {
         
         //other cases are tested in testComparison
     }
+    
+    @Test
+    public void testReplacement() throws IOException {
```

Comment:
Ok, it's good that the issue at hand is understood better now.

1. Yes, it's true that `verifyHtml` does all the steps in one go, however do remember that `verifyHtml` works _with a real web page seen on a browser_ and already has an abstraction on which part of the HTML page source is examined (mainContent, chosen part, entire file, ...). This test, on the other hand, is purely to test how the injection/replacement of values work.
2. With regards to "this test adds no value (other than the fact that it would be run more often)", that's exactly the value I'm looking for, and back to the second reason I mentioned: this algorithm is used _at all times_ instead of only during regeneration, and something that is used _at all times_ should be tested _at all times_ instead of only occasionally (real use case: if we suspect the algorithm to be broken, would you rather run `GodModeTest` just for the sake of checking if it is really broken, or having `HtmlHelperTest` catching it concurrently with the whole test suite?).

In other words, the main purpose of each test:
`HtmlHelperTest`: testing the injection and replacement of values
`GodModeTest`: testing the regeneration of HTML file

They are very different. However, I won't deny that in the current state `GodModeTest` also serves the purpose of `HtmlHelperTest` mentioned. I will agree that this test is no longer needed if we can integrate `GodModeTest` back to the full test suite.